phoenicis-javafx/src/main/java/org/phoenicis/javafx/views/mainwindow/library/LibrarySideBar.java

```diff
@@ -0,0 +1,145 @@
+package org.phoenicis.javafx.views.mainwindow.library;
+
+import javafx.scene.layout.VBox;
+import javafx.stage.FileChooser;
+import org.phoenicis.javafx.views.mainwindow.ui.*;
+import org.phoenicis.library.dto.ShortcutDTO;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.function.Consumer;
+
+import static org.phoenicis.configuration.localisation.Localisation.translate;
+
+/**
+ * Created by marc on 15.04.17.
+ */
+public class LibrarySideBar extends VBox {
+    // the search bar used for filtering
+    private SearchBox searchBar;
+
+    // the shortcut group, containing the run, stop and uninstall buttons
+    private LeftGroup shortcutGroup;
+
+    private LeftButton runButton;
+    private LeftButton stopButton;
+    private LeftButton uninstallButton;
+
+    // the advanced tools group, containing the run script and run console buttons
+    private LeftGroup advancedToolsGroup;
+
+    private LeftButton runScript;
+    private LeftButton runConsole;
+
+    // the name of this application
+    private final String applicationName;
+
+    // consumer called after a text has been entered in the search box
+    private Consumer<String> onSearch;
+
+    // the current selected shortcut
+    private ShortcutDTO shortcut;
+
+    // consumers called when a shortcut should be run, stopped or uninstalled
+    private Consumer<ShortcutDTO> onShortcutRun;
+    private Consumer<ShortcutDTO> onShortcutStop;
+    private Consumer<ShortcutDTO> onShortcutUninstall;
+
+    // consumers called when a script should be run or a console be opened
+    private Consumer<File> onScriptRun;
+    private Runnable onOpenConsole;
+
+    public LibrarySideBar(String applicationName) {
```

Comment:
Which methods do you mean? 
The callbacks or the methods inside the class?
About the callbacks: I myself don't know the use of all the callbacks :/